python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -287,11 +290,241 @@ def _fit(model, X_df_list, y_df_list, r):
             y_df = cudf.concat(y_df_list)
+        #print(model.print_summary())
         return model.fit(X_df, y_df)
 
     @staticmethod
-    def _predict(model, X, r):
-        return model._predict_get_all(X)
+    def _predict(model, X, treelite_handle):
+        #print(model.print_summary())
+        #import pdb
+        #print("length of X_df_list is : ", len(X_df_list))
+        #pdb.set_trace()
+        #if len(X_df_list) == 1:
+        #    X_df = X_df_list[0]
+        #else:
+        #    X_df = cudf.concat(X_df_list)
+        #pdb.set_trace()
+
+        return model._predict_mnmg_on_fil(X, treelite_handle=treelite_handle) #.copy_to_host()
+
+    @staticmethod
+    def _print_summary(model):
+        model.print_summary()
+
+    @staticmethod
+    def _tl_model_handles(model, model_bytes):
+        #print("model info inside _convert_to_tl : ", model_info)
+        return model._tl_model_handles(model_bytes=model_bytes)
+
+    @staticmethod
+    def _get_model_info(model):
+        return model._get_model_info()
+
+    @staticmethod
+    def _build_fil_model(model, model_handle):
+        print("model handels in dask func : ", model_handle)
+        return model._build_fil_model(model_handle=model_handle)
+
+    @staticmethod
+    def _read_mod_handles(model, mod_handles):
+        return model._read_mod_handles(mod_handles=mod_handles)
+
+    def get_model_info(self):
+        """
+        get the model information, convert it to model bytes
+        """
+        c = default_client()
+        futures = list()
+        workers = self.workers
+        for n, w in enumerate(workers):
+            futures.append(
+                c.submit(
+                    RandomForestClassifier._get_model_info,
+                    self.rfs[w],
+                    workers=[w],
+                )
+            )
+
+        wait(futures)
+        raise_exception_from_futures(futures)
+
+        ### this was for concatenating the model bytes obtained when converting
+        ### cuML -> TL -> model_bytes
+        print("len of futures : ", len(futures))
+        mod_bytes = list()
+        len_mod_bytes = list()
+        for i in range(len(futures)):
+            mod_bytes.append(futures[i].result())
+
+        #self.concat_model_bytes = list(chain.from_iterable(mod_bytes))
+
+        return mod_bytes #self.concat_model_bytes
+
+    def print_summary(self):
+        """
+        prints the summary of the forest used to train and test the model
+        """
+        c = default_client()
+        futures = list()
+        workers = self.workers
+
+        for n, w in enumerate(workers):
+            futures.append(
+                c.submit(
+                    RandomForestClassifier._print_summary,
+                    self.rfs[w],
+                    workers=[w],
+                )
+            )
+
+        wait(futures)
+        raise_exception_from_futures(futures)
+        return self
+
+    def convert_to_treelite(self):
```

Comment:
It looks like this whole file is about converting an MNMG cuML RF model into treelite. I think it would be better to move it to a separate pull request.